[Thaw Tun Zan] iP

ThawTunZan opened the pull request.

ThawTunZan committed:
Duke.java: replace original greeting with ThawBot greeting

ThawTunZan committed:
Add Echo for level 1 Increment
This commit does the following:
*Adds a feature where the program repeat the user's input
*Adds a feature that will allow the user to manually exit the program

ThawTunZan committed:
Edit Echo by adding List for level 2 increment
This commit does the following:
*Adds a new command "list" that displays all the previously typed texts by the user
*Adds a feature where every line of texts entered by user is stored

ThawTunZan committed:
Add "mark as done" feature for level 3 increment
Add Task class that replaces the original list of strings that contains the user inputs

ThawTunZan committed:
Level 4 Increment
Add support for tracking three types of tasks:
ToDos: tasks without any date/time attached to it e.g., visit new theme park
Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific date/time and ends at a specific date/time

NicholasTanYY commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Override
+    public String getStatusIcon() {
+        return (isDone ? "[D][X] " + super.description + " (by: " + by + ")": "[D][ ] " + super.description + " (by: " + by + ")"); // mark done task with X
```
this line is a bit long! Consider splitting this into 2 lines?

NicholasTanYY commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+
+    }
+
+    private static void printsGreeting() {
```
i think we can use a singular verb here, to print the greetings
```suggestion
    private static void printGreeting() {
```

NicholasTanYY commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+            int startIndex = usersInput.indexOf("from");
+            int secondIndex = usersInput.indexOf("to");
+            list[currentIteration] = new Event(usersInput.substring(0, startIndex - 2),
+                    "from: " + usersInput.substring(startIndex+ 5, secondIndex - 2) + " " + "to: " + usersInput.substring(secondIndex + 3));
```
this line is another long one! consider splitting into 2 lines?

NicholasTanYY commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Override
+    public String getStatusIcon() {
+        return (isDone ? "[E][X] " + super.description + " (" + event + ")": "[E][ ] " + super.description + " (" + event + ")"); // mark done task with X
```
another long line! consider splitting into 2 lines

NicholasTanYY reviewed (commented):
Good job on maintaining most of the coding standards! Overall, I think that your code was neat. There were a few instances of long lines of code that may not exactly have exceeded the 120 chars line limit, but the code may be neater when these long lines are split into more lines.

ThawTunZan committed:
fix: bug#1
Bug is printing the type of task like todo, event and deadline when the bot respond to you after adding a task.
E.g. "[][] todo read book" instead of "'[][] read book

nicknamenic commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+    }
+
+    private static void printGoodByeMessage() {
+        String goodbyeMessage = "Bye. Hope to see you again soon!";
```
A bit redundant compared to just printing out the string.

nicknamenic commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+        System.out.println(greetingMessage);
+    }
+
+    private static void printGoodByeMessage() {
```
This persists throughout, but even though a lot of the methods are small, it's a good choice to make them in order to give a sense of clarity to future methods and especially main.

nicknamenic commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+        }
+    }
+
+    private static void markOrUnmarkTask(String usersInput, boolean isDone, Task[] task) {
```
Method and variable names are pretty clear in general, which is nice to see.

nicknamenic commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+        int currentIteration = 0;
+        boolean canExit = false;
+
+        while (!canExit) {
```
Very efficient code! In my opinion it's hard to make this method not have a lot of nested conditions or loops.

nicknamenic commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+                canExit = true;
+                printGoodByeMessage();
+            } else if (usersInput.equals("list")) {
+                printList(currentIteration, list);
```
Could use a few more comments here and there especially for the longer methods, but otherwise the code is very clear and easy to understand.

ThawTunZan committed:
Add Level 5 increment
This commit contains the error exception handling for invalid user inputs

ThawTunZan committed:
Reduce code length for most of the functions

ThawTunZan committed:
Merge branch 'branch-Level-5'

ThawTunZan committed:
Use Java Collection class ArrayList

ThawTunZan committed:
Add support for deleting tasks from the list

ThawTunZan committed:
Add file loading when bot comes online

ThawTunZan committed:
Implement loading from saved data

ThawTunZan committed:
Add level-7 increment
Implement a save feature

ThawTunZan committed:
Merge branch 'branch-Level-6'

ThawTunZan committed:
Change ThawBot.txt

ThawTunZan committed:
Add ThawBot.txt for save load

ThawTunZan committed:
Remove content in ThawBot.txt

ThawTunZan committed:
Merge branch 'branch-Level-7'

# Conflicts:
Conflicts between branch 7 and 6 with the use of ArrayList and Array
#	src/main/java/Duke.java

ThawTunZan committed:
Update gitignore to remove ThawBot.txt from version control

ThawTunZan committed:
Refactor classes into 5 packages (for readability):
* Commands package
* Exception package
* Filemanager package
* PrintMessages package
* Tasks package

ThawTunZan committed:
Restructure the way exceptions are handled
Exception is throw from the classes in the commands package instead of from main, making the code looks neater

ThawTunZan committed:
Make the code more OOP
extract the following classes:
*UI: For displaying response messages from the bot
*Storage: Saving and loading of files
*Parser: Process user's input
*TaskList: Contains the task and its operations

ThawTunZan committed:
Level 8 increment
Store deadline date as both the java.time.LocalDate format and a string format

ThawTunZan committed:
Level 9 Increment
Give users a way to find a task by searching for a keyword

ThawTunZan committed:
Merge pull request #2 from ThawTunZan/branch-Level-8

Level 8 increment

ThawTunZan committed:
A JavaDoc increment

ThawTunZan committed:
Merge branch 'master' into branch-A-JavaDoc

ThawTunZan committed:
Merge branch 'master' into branch-Level-9

ThawTunZan committed:
Merge pull request #3 from ThawTunZan/branch-A-JavaDoc

Branch a java doc

ThawTunZan committed:
Merge branch 'master' into branch-Level-9

ThawTunZan committed:
Merge pull request #4 from ThawTunZan/branch-Level-9

Branch level 9

ThawTunZan committed:
Add three new class that inherits from Command class
*AddToDoTask
*AddEventTask
*AddDeadlineTask
*Add more documentation

ThawTunZan committed:
Add documentation for most of the classes

ThawTunZan committed:
fix: bug#2  After adding a task the scanner doesn't read the user's input from a new line instead it reads from the same sentence as what was previously printed

ThawTunZan committed:
Update README.md

ThawTunZan committed:
Update README.md

ThawTunZan committed:
Update README.md

ThawTunZan committed:
Update README.md

ThawTunZan committed:
Update README.md

ThawTunZan committed:
fix: bug#3 Bugs with print statements and adding of event task
*Event task not added
*Wrong print statement for unmark command

ThawTunZan committed:
Update README.md

ThawTunZan committed:
Update README.md

ThawTunZan committed:
Update README.md

ThawTunZan committed:
update gitignore

ThawTunZan committed:
Update README.md

ThawTunZan committed:
Remove magic numbers

ThawTunZan committed:
Remove unecessary imports
